edb/schema/functions.py

```diff
@@ -1251,6 +1252,119 @@ def get_dummy_body(self, schema: s_schema.Schema) -> s_expr.Expression:
+            multi_overload = False
+
+            for i, (new_t, ext_t) in enumerate(zip(new_pt, ext_pt)):
+                if new_t != ext_t:
+                    if new_t.is_object_type() and ext_t.is_object_type():
+                        if (
+                            this_diff_param != -1
+                            or (
+                                diff_param != -1
+                                and diff_param != this_diff_param
+                            )
+                            or non_obj_param_diff
+                        ):
+                            multi_overload = True
+                            break
+                        else:
+                            this_diff_param = i
+                    else:
+                        non_obj_param_diff = True
+                        if this_diff_param != -1:
+                            multi_overload = True
+                            break
+
+            if this_diff_param != -1:
+                if not multi_overload:
+                    multi_overload = len(new_params) != len(ext_params)
+
+                if multi_overload:
+                    # Multiple dispatch of object-taking functions is
+                    # not supported.
+                    my_sig = self.get_signature_as_str(schema)
+                    other_sig = f.get_signature_as_str(schema)
+                    raise errors.UnsupportedFeatureError(
+                        f'cannot create the `{my_sig}` function: '
+                        f'overloading an object type-receiving '
+                        f'function with differences in the remaining '
+                        f'parameters is not supported',
+                        context=srcctx,
+                        details=(
+                            f"Other function is defined as `{other_sig}`"
+                        )
+                    )
+
+                if not all(
+                    new_p.get_parameter_name(schema)
+                    == ext_p.get_parameter_name(schema)
+                    for new_p, ext_p in zip(new_params, ext_params)
+                ):
+                    # And also _all_ parameter names must match due to
+                    # current implementation constraints.
+                    my_sig = self.get_signature_as_str(schema)
+                    other_sig = f.get_signature_as_str(schema)
+                    raise errors.UnsupportedFeatureError(
+                        f'cannot create the `{my_sig}` '
+                        f'function: overloading an object type-receiving '
+                        f'function with differences in the names of '
+                        f'parameters is not supported',
+                        context=srcctx,
+                        details=(
+                            f"Other function is defined as `{other_sig}`"
+                        )
+                    )
+
+                diff_param = this_diff_param
+                overloads.append(f)
```

Comment:
You can only check against the first overload and then break the top loop.

Reply:
I can't: I need all matching overloads to construct the common function body.